app/controllers/referee_interface/reference_controller.rb

```diff
@@ -0,0 +1,47 @@
+module RefereeInterface
+  class ReferenceController < ActionController::Base
+    layout 'application'
+
+    def feedback
+      reference = Reference.find_by(token: params[:token])
```

Comment:
If you do `Reference.find_by!` it'll trigger an `RecordNotFound` exception that you can catch with a `rescue_from`.